cylc/flow/option_parsers.py

```diff
@@ -842,7 +854,11 @@ def cleanup_sysargv(
     # replace source path with workflow ID.
     if str(source) in sys.argv:
         sys.argv.remove(str(source))
-    if workflow_id not in sys.argv:
+    # Add workflow ID if it's not already present.
+    if (
+        workflow_id not in sys.argv
+        and workflow_id.split('/')[0] not in sys.argv
```

Comment:
This will presumably fail for hierarchical workflows e.g. `a/b/c/d/run1`?

Reply:
I thought that this was simplistic.

Reply:
Perhaps it's easier to solve this issue in the `cylc vip` script?

Reply:
Yes, it was easier to solve in VR. 

I've also now got a good explanation of why the problem doesn't occur in VIP - `parse_id` is called by VR because it needs to parsed id to work out whether the workflow is running or stopped. VIP does not do this so the id passed to `cleanup_sysargv` is always the same as the one in `sys.argv`.